ui/src/app/shared/requirements/form/requirements.form.component.ts

```diff
@@ -28,40 +28,55 @@ export class RequirementsFormComponent {
 
     newRequirement: Requirement = new Requirement('binary');
     availableRequirements: Array<string>;
-    valueChanged = false;
-    workerModels: Array<string>;
+    workerModels: Array<WorkerModel>;
     _suggest: Array<string> = [];
     loading = true;
 
-    constructor(private _requirementStore: RequirementStore, private _workerModelService: WorkerModelService) {
+    constructor(private _requirementStore: RequirementStore,
+        private _workerModelService: WorkerModelService,
+        private _translate: TranslateService) {
         this._requirementStore.getAvailableRequirements().subscribe(r => {
             this.availableRequirements = new Array<string>();
-            this.availableRequirements.push(...r.toArray());
+            // user does not need to add plugin prequisite manually, so we remove it from list
+            this.availableRequirements.push(...r.filter(req => req !== 'plugin').toArray());
         });
 
         this._workerModelService.getWorkerModels().first()
         .finally(() => this.loading = false)
         .subscribe( wms => {
-            this.workerModels = wms.map((wm) => wm.name).concat(this.workerModels);
+            this.workerModels = wms;
         });
     }
 
-    addRequirement(): void {
-        this.event.emit(new RequirementEvent('add', this.newRequirement));
-        this.newRequirement = new Requirement('binary');
-        this.valueChanged = false;
+    onSubmitAddRequirement(form): void {
+        if (form.valid === true && this.newRequirement.name !== '' && this.newRequirement.value !== '') {
```

Comment:
maybe reflect this condition for a `disabled` class on add button ?

Reply:
done